Gives an animation and a noise when throwing something

Tristrian opened the pull request:
## About The Pull Request
Whenever a carbon mob (players and monkeys for example) throws something, it makes a small motion forward similar when striking someone. It also makes a noise when throwing something.
https://streamable.com/ikjytk?src=player-page-share
The sound, volume, and pitch will change depending on the weight of the item, the thrown distance, and your power levels for throwing. 

Also fixes a small oddity in chat saying that dwarves were throwing "really hard" things despite being weaker than a normal human.

This was noticed from a downstream (Skyrat) and I thought this could be something added upstream.
The original credits go to [Gandalf2k15](https://github.com/Gandalf2k15) and [Azarak](https://github.com/Azarak)
## Why It's Good For The Game
Currently, when you throw something, your sprite stands still, silent, and motionless, as an item suddenly appears from _somewhere_. Depending on how fast the projectile is, you can't see who has thrown it until you check the chat.

Making your sprite motion when throwing gives a better feedback on who has thrown something while also being aesthethically nice. If punching things makes your sprite move and make noise, then throwing could also have this same visual.
## Changelog
:cl:
image: Throwing things has now a sprite animation forward your target.
sound: Throwing things also has a sound accompagnying it.
spellcheck: Dwarves don't throw things hard like hulks. Instead their throws are described as "flimsily".
/:cl:

Tristrian committed:
adds a little animation with an accompagnying noise with the thrower.

MrMelbert commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -171,6 +171,8 @@
 		if(thrown_item.throw_verb)
 			verb_text = thrown_item.throw_verb
+	do_attack_animation(target, no_effect = 1)
+	playsound(loc, 'sound/weapons/punchmiss.ogg', 50, TRUE, -1)
```
```suggestion
	playsound(src, 'sound/weapons/punchmiss.ogg', 50, vary = TRUE, extrarange = -1)
```
- `src` or `loc` doesn't matter, the former is more clear though
- named args

MrMelbert replied:
Also just a thought but I wonder if pitching the sound effect based on how strong the throw is (power and range) would be neat.

Tristrian replied:
I will try fiddling with the values and see if there is a satisfying setup.

GoldenAlpharex reviewed (approved):
Would just be nice to credit where you took that from (since it was originally added on Skyrat, but I have no clue if they had taken it from somewhere else themselves).

optimumtact commented:
Yeah you really should link back to at a minimum the PR that added this to the downstream.

Tristrian committed:
Update code/modules/mob/living/carbon/carbon.dm

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Tristrian commented:
I have finally managed to find the authors, thanks Golden for showing me how to : 
https://github.com/Skyrat-SS13/Skyrat-tg/pull/1159/
https://github.com/Skyrat-SS13/Skyrat-tg/pull/1367
The original credits go to [Gandalf2k15](https://github.com/Gandalf2k15) and [Azarak](https://github.com/Azarak)
I will attempt to change the pitch of the items depending on range and strength like requested.

Tristrian commented:
> If its the same noise Skyrat uses, beware that, in my experience, it gets old quick. Maybe add some more varying throw sounds?

It's currently the same sound as missing a punch. Altough I am trying to find a way to add further variant depending on the range and strength. 
A short throw would have a lower volume for example.

Tristrian committed:
Update carbon.dm

Tristrian commented:
I will think more about it tomorrow. But so far, the ideas are : 

- The volume of the throw will adjust to the range. A short throw will be more subtle in noise that throwing across the hallway.
- The speed of the throw will adjust the pitch. Fast throws will sound sharper.
- Changing the type of sound if it's a power throw or not. I am not sure which current sound file would be fitting for a hulk power throw so I am open to suggestions.

Tristrian committed:
Add three new sounds for throwing

3 new sounds depending on the throw strength (hulk, dwarf, etc...)
Source of the sounds : 
throw.ogg : https://uppbeat.io/sfx/throwing-item-swing/9430/25267
throwhard.ogg : https://uppbeat.io/sfx/whoosh-air-punch/114/1168
throwsoft.ogg : https://uppbeat.io/sfx/whoosh-swift-cut/7727/23617

Tristrian committed:
Adds throw sound variants and corrects some text.

Hulks use a loud and heavier sound.
Dwarves use a slower and lazier sound.
I switched the throw.ogg and throwsoft.ogg in between.
Also dwarves no longer throw hard.

Tristrian committed:
Final polishing and adjustements

Had to set the default frequency upper, because thrown mobs don't have a w_class and it broke throwing them.
Mobs can be thrown again.
Also adjusted the volume of the throwhard.ogg and throwsoft.ogg to match more with the default throw.

Tristrian commented:
After fiddling around and testing, I think I have found a decent sound distribution.

- 3 types of sound, depending on your power_throw. Hulk and hard throws will sound heavier and meatier. While negative power_throws made by dwarves sound lazier and slower.
- The volume of the sound depends on the distance the item is thrown. Longer throws are louder. There is a cap to not be deafening at least.
- The weight of the item changes the pitch of the sound. Tiny items will sound sharper, huge items will sound heavier. It was intended first to depend on the throw speed, but it was too inconsistent. A smoking pipe sounded like you were throwing a truck compared to a dice.

Here is an example in game from someone throwing things around you, to test the different volume levels : https://streamable.com/ikjytk

Let me know if there needs more adjustements.

Source of the sounds : 
throw.ogg : https://uppbeat.io/sfx/whoosh-swift-cut/7727/23617
throwhard.ogg : https://uppbeat.io/sfx/whoosh-air-punch/114/1168
throwsoft.ogg :  https://uppbeat.io/sfx/throwing-item-swing/9430/25267

Ghommie commented:
Hey, could you include the sources of the sounds in the attributions text file inside the sounds folder? Thanks.

Tristrian committed:
Update attributions.txt

Tristrian commented:
> Hey, could you include the sources of the sounds in the attributions text file inside the sounds folder? Thanks.

Done, let me know if there is something else missing or done wrong.
